### PR TITLE
#3501 Pagination for global offers for Basilisk

### LINKED ISSUE #3501: Pagination for global offers for Basilisk
Let's introduce pagination for global offers.

By default, I would let it show 20 items so it can fit a 13" notebook by default? 

https://user-images.githubusercontent.com/5887929/180041427-21cfc889-fb37-4d65-8c15-3b37771e2f5c.mov

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3501
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2022-07-20 at 6 17 29 PM

### CODE REVIEW

**Review by petersopko — COMMENTED** [review]
wfm

**Review by petersopko — APPROVED** [review]

**Comment by Jarsen136** [comment]
Let's resolve this conflicts and it's good to go

**Comment by prachi00** [comment]
> Let's resolve this conflicts and it's good to go

done

File: components/bsx/Offer/OfferTable.vue [hunk 1]
```diff
@@ -1,114 +1,123 @@
...
+      enableListenKeyboardEvent
+      preserveScroll />
+    <b-table :data="showList">
+      <b-table-column
+        v-if="isBsxStats"
+        cell-class="is-vcentered is-narrow"
+        :label="$t('offer.collection')"
+        v-slot="props"
+        sortable>
+        <nuxt-link :to="`collection/${props.row.nft.collection.id}`">
```
Comment (Jarsen136) [inline components/bsx/Offer/OfferTable.vue:17]:
```suggestion
      <nuxt-link :to="`/bsx/collection/${props.row.nft.collection.id}`">
```

File: components/bsx/Offer/OfferTable.vue [hunk 2]
```diff
@@ -1,114 +1,123 @@
...
+          </p>
+        </nuxt-link>
+      </b-table-column>
+      <b-table-column
+        v-if="isBsxStats"
+        cell-class="is-vcentered is-narrow"
+        :label="$t('offer.nftName')"
+        v-slot="props"
+        sortable>
+        <nuxt-link :to="`gallery/${props.row.nft.id}`">
```
Comment (Jarsen136) [inline components/bsx/Offer/OfferTable.vue:39]:
```suggestion
        <nuxt-link :to="`gallery/${props.row.nft.id}`">
```
      <nuxt-link :to="`/bsx/gallery/${props.row.nft.id}`">

**Review by Jarsen136 — COMMENTED** [review]
small stuff from conflict code

File: components/bsx/Offer/OfferTable.vue (hunk 2 above)
Comment (Jarsen136) [reply components/bsx/Offer/OfferTable.vue]: /bsx/gallery

Comment (prachi00) [reply components/bsx/Offer/OfferTable.vue]: @Jarsen136 I did change it, if you look at the file its updated

Comment (Jarsen136) [reply components/bsx/Offer/OfferTable.vue]: no, I see it lack of '/' at the beginning

File: components/bsx/Offer/OfferTable.vue [hunk 3]
```diff
@@ -1,114 +1,123 @@
...
+          </p>
+        </nuxt-link>
+      </b-table-column>
+      <b-table-column
+        v-if="isBsxStats"
+        cell-class="is-vcentered is-narrow"
+        :label="$t('offer.nftName')"
+        v-slot="props"
+        sortable>
+        <nuxt-link :to="`bsx/gallery/${props.row.nft.id}`">
```
Comment (Jarsen136) [inline components/bsx/Offer/OfferTable.vue:39]:
```suggestion
        <nuxt-link :to="`/bsx/gallery/${props.row.nft.id}`">
```

**Review by Jarsen136 — COMMENTED** [review]
pls take a look : )

File: components/bsx/Offer/OfferTable.vue [hunk 4]
```diff
@@ -137,6 +152,35 @@ export default class OfferTable extends mixins(OfferMixin) {
...
+      return 'computing'
+    }
+    if (this.currentBlock > expirationBlock) {
+      return 'expired'
+    }
+    const secondsForEachBlock = 12
+    const diffSeconds =
+      secondsForEachBlock * (expirationBlock - this.currentBlock)
+    return formatSecondsToDuration(diffSeconds)
+  }
```
Comment (Jarsen136) [inline components/bsx/Offer/OfferTable.vue:183]: pls delete `created()` and  `calcExpirationTime(expirationBlock: number)` function.
They are already in `OfferMixin`

**Review by Jarsen136 — COMMENTED** [review]
The last one comment

File: components/bsx/Offer/OfferTable.vue (hunk 2 above)
Comment (prachi00) [reply components/bsx/Offer/OfferTable.vue]: oh oops, added now

File: components/bsx/Offer/OfferTable.vue (hunk 4 above)
Comment (prachi00) [reply components/bsx/Offer/OfferTable.vue:183]: oh okay

**Comment by Jarsen136** [comment]
There are some new conflicts.

**Comment by prachi00** [comment]
> There are some new conflicts.

😭 why is this happening to me

**Comment by prachi00** [comment]
> > There are some new conflicts.
> 
> 😭 why is this happening to me

doneee

**Review by Jarsen136 — APPROVED** [review]
LGTM

**Comment by Jarsen136** [comment]
pay 75

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $75 @ 65.34 USD/KSM ~ 1.148 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0x7f36a4916c96f844476d4d7d513268f6388c7e54894d9e5058ad7d0aaa95c6d3](https://kusama.subscan.io/extrinsic/0x7f36a4916c96f844476d4d7d513268f6388c7e54894d9e5058ad7d0aaa95c6d3)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**